File: api/app/models/usage.py
from datetime import date

from pydantic import BaseModel


class UsageEntry(BaseModel):
    name: str
    percent: float


class PokemonUsageResponse(BaseModel):
    id: int
    pokemon_name: str
    format: str
    usage_percent: float
    moves: list[UsageEntry] | None = None
    items: list[UsageEntry] | None = None
    abilities: list[UsageEntry] | None = None
    teammates: list[UsageEntry] | None = None
    snapshot_date: date
    source: str | None = None
    sprite_url: str | None = None


class PokemonUsageList(BaseModel):
    data: list[PokemonUsageResponse]
    count: int

File: web/src/types/usage.ts
export interface UsageEntry {
  name: string;
  percent: number;
}

export interface PokemonUsage {
  id: number;
  pokemon_name: string;
  format: string;
  usage_percent: number;
  moves: UsageEntry[] | null;
  items: UsageEntry[] | null;
  abilities: UsageEntry[] | null;
  teammates: UsageEntry[] | null;
  snapshot_date: string;
  source: string | null;
  sprite_url: string | null;
}

export interface PokemonUsageList {
  data: PokemonUsage[];
  count: number;
}
